test/account_api_test.rb

```diff
@@ -0,0 +1,407 @@
+        .with(
+          method: :post,
+          path: "/api/oauth2/callback",
+          body: Pact.like(code: "code", state: "state"),
+          headers: GdsApi::JsonClient.default_request_with_json_body_headers,
+        )
+        .will_respond_with(
+          status: 401,
+          headers: { "Content-Type" => "application/json; charset=utf-8" },
+        )
+    end
+
+    it "responds with 401 Unauthorized" do
+      assert_raises GdsApi::HTTPUnauthorized do
+        api_client.validate_auth_response(code: "code", state: "state")
+      end
+    end
+  end
+
+  describe "creating a registration state" do
+    let(:attributes) { { foo: "bar" } }
+
+    before do
+      account_api
+        .given("no particular state")
+        .upon_receiving("a create-state request")
+        .with(
+          method: :post,
+          path: "/api/oauth2/state",
+          body: { attributes: attributes },
+          headers: GdsApi::JsonClient.default_request_with_json_body_headers,
+        )
+        .will_respond_with(
+          status: 200,
+          headers: { "Content-Type" => "application/json; charset=utf-8" },
+          body: {
+            state_id: Pact.like("string"),
+          },
+        )
+    end
+
+    it "responds with 200 OK and a state_id" do
+      response = api_client.create_registration_state(attributes: attributes)
+      assert response["state_id"].present?
+      assert_equal 200, response.code
+    end
+  end
+
+  describe "checking for a transition checker email subscription" do
+    describe "the user is logged in" do
+      let(:given) { "there is a valid user session" }
+      let(:has_subscription) { false }
+
+      before do
+        account_api
+          .given(given)
+          .upon_receiving("a has-subscription request")
+          .with(
+            method: :get,
+            path: "/api/transition-checker-email-subscription",
+            headers: GdsApi::JsonClient.default_request_headers.merge(authenticated_headers),
+          )
+          .will_respond_with(
+            status: 200,
+            headers: { "Content-Type" => "application/json; charset=utf-8" },
+            body: {
+              govuk_account_session: Pact.like("string"),
+              has_subscription: has_subscription,
+            },
+          )
+      end
+
+      it "responds with 200 OK, a new govuk_account_session, and says that the subscription does not exists" do
+        response = api_client.check_for_email_subscription(govuk_account_session: govuk_account_session)
+        assert response["govuk_account_session"].present?
+        assert_equal false, response["has_subscription"]
+        assert_equal 200, response.code
+      end
+
+      describe "a subscription does not exist" do
+        let(:given) { "there is a valid user session, with a transition checker email subscription" }
+        let(:has_subscription) { true }
+
+        it "says that the subscription does not exist" do
```

Comment:
Is that what's being asserted below? 🤔

Reply:
Whoops, I flipped the logic of this one and didn't update the name